src/requestSchemas.ts

```diff
@@ -0,0 +1,169 @@
+  'manifest',
+  'include-components'
+];
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#requirements
+const UNSUPPORTED_DATA_REQ_ARGS = [
+  'expression',
+  'parameters',
+  'manifest',
+  'include-dependencies',
+  'include-components'
+];
+
+// Operation Definition: https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#search
+const UNSUPPORTED_SEARCH_ARGS = [
+  'date',
+  'effective',
+  'jurisdiction',
+  'context',
+  'context-type',
+  'context-type-quantity',
+  'context-type-value',
+  'topic',
+  'composed-of',
+  'depends-on',
+  'derived-from',
+  'successor',
+  'predecessor'
+];
+
+const customErrorMap: z.ZodErrorMap = (issue, ctx) => {
+  if (issue.code === z.ZodIssueCode.custom) {
+    if (issue.params?.type === 'NotImplemented') {
+      throw new NotImplementedError(issue.message ?? ctx.defaultError);
+    } else {
+      throw new BadRequestError(issue.message ?? ctx.defaultError, issue.params?.serverErrorCode);
+    }
+  } else if (issue.code === z.ZodIssueCode.unrecognized_keys) {
+    throw new BadRequestError(issue.message ?? ctx.defaultError, constants.ISSUE.CODE.VALUE);
+  }
+  throw new BadRequestError(issue.message ?? ctx.defaultError);
+};
+
+z.setErrorMap(customErrorMap);
+
+const hasIdentifyingInfo = (args: Record<string, any>) => args.id || args.url || args.identifier;
+
+/**
+ * Returns a function that checks if any unsupported params are present, then runs the
+ * other passed in functions in sequence. Each catchFunction is expected to check for
+ * invalid input and call ctx.addIssue if invalid input is detected. Any added issue will
+ * trigger an error to be thrown.
+ */
+export function catchInvalidParams(
+  catchFunctions: ((val: Record<string, any>, ctx: z.RefinementCtx) => void)[],
+  unsupportedArgs?: string[]
+) {
+  return (val: Record<string, any>, ctx: z.RefinementCtx) => {
+    const includedUnsupported = Object.keys(val).filter((v: string) => unsupportedArgs?.includes(v));
+    if (includedUnsupported.length > 0) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        params: { type: 'NotImplemented' },
+        message: `Parameters: ${includedUnsupported.join(', ')} are not yet supported on this server`
+      });
+    } else {
+      catchFunctions.forEach(func => func(val, ctx));
+    }
+  };
+}
+
+/**
+ * Checks that either id, url, or identifier parameter is included. Adds an issue to the ctx
+ * that triggers a BadRequest to be thrown if not.
+ */
+export function catchMissingIdentifyingInfo(val: Record<string, any>, ctx: z.RefinementCtx) {
+  if (!hasIdentifyingInfo(val)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      params: { serverErrorCode: constants.ISSUE.CODE.REQUIRED },
+      message: 'Must provide identifying information via either id, url, or identifier parameters'
+    });
+  }
+}
+
+const stringToBool = z.enum(['true', 'false']).transform(x => x === 'true');
```

Comment:
It actually doesn't because `z.coerce.boolean()` just calls `Boolean(val)`, so if you sent in `false` it would actually evaluate to `true` since strings are truthy.

Reply:
thanks I hate it